src/main/java/org/opensearch/knn/index/codec/nativeindex/remote/RemoteIndexBuildStrategy.java

```diff
@@ -146,13 +146,14 @@ public void buildAndWriteIndex(BuildIndexParams indexInfo) throws IOException {
 
             // 4. Download index file and write to indexOutput
             readFromRepository(indexInfo, repositoryContext, remoteBuildStatusResponse);
-
             success = true;
+            return;
         } catch (Exception e) {
-            fallbackStrategy.buildAndWriteIndex(indexInfo);
+            log.warn("Failed to build index remotely", e);
```

Comment:
exception message here is incomplete. It doesn't provide any information about for what parameters index build failed. Lets log all of them to understand what failed.

Another thing is why this is warm log? This should be a error log.

Reply:
It is always better to have those parameters. I would suggest logging them. So lets add it.